core/playlistmanager.go

```diff
@@ -64,55 +64,34 @@ func (mgr *BasicPlaylistManager) GetOSSession() drivers.OSSession {
 	return mgr.storageSession
 }
 
-func (mgr *BasicPlaylistManager) isRightStream(streamID StreamID) error {
-	mid := streamID.ManifestID
-	if mid != mgr.manifestID {
-		return fmt.Errorf("Wrong manifest id (%s), should be %s", mid, mgr.manifestID)
-	}
-	return nil
-}
-
-func (mgr *BasicPlaylistManager) hasSegment(mpl *m3u8.MediaPlaylist, seqNum uint64) bool {
-	for _, seg := range mpl.Segments {
-		if seg != nil && seg.SeqId == seqNum {
-			return true
-		}
-	}
-	return false
-}
-
-func (mgr *BasicPlaylistManager) getPL(streamID StreamID) *m3u8.MediaPlaylist {
+func (mgr *BasicPlaylistManager) getPL(rendition string) *m3u8.MediaPlaylist {
 	mgr.mapSync.RLock()
-	mpl := mgr.mediaLists[streamID]
+	mpl := mgr.mediaLists[rendition]
 	mgr.mapSync.RUnlock()
 	return mpl
 }
 
-func (mgr *BasicPlaylistManager) createPL(streamID StreamID) *m3u8.MediaPlaylist {
+func (mgr *BasicPlaylistManager) createPL(profile *ffmpeg.VideoProfile) *m3u8.MediaPlaylist {
 	mpl, err := m3u8.NewMediaPlaylist(LIVE_LIST_LENGTH, LIVE_LIST_LENGTH)
 	if err != nil {
 		glog.Error(err)
 		return nil
 	}
 	mgr.mapSync.Lock()
-	mgr.mediaLists[streamID] = mpl
+	mgr.mediaLists[profile.Name] = mpl
 	mgr.mapSync.Unlock()
-	sid := streamID.String()
-	vParams := ffmpeg.VideoProfileToVariantParams(ffmpeg.VideoProfileLookup[streamID.Rendition])
-	url := sid + ".m3u8"
+	vParams := ffmpeg.VideoProfileToVariantParams(*profile)
+	url := fmt.Sprintf("%v/%v.m3u8", mgr.manifestID, profile.Name)
 	mgr.masterPList.Append(url, mpl, vParams)
 	return mpl
 }
 
-func (mgr *BasicPlaylistManager) InsertHLSSegment(streamID StreamID, seqNo uint64, uri string,
+func (mgr *BasicPlaylistManager) InsertHLSSegment(profile *ffmpeg.VideoProfile, seqNo uint64, uri string,
 	duration float64) error {
 
-	if err := mgr.isRightStream(streamID); err != nil {
-		return err
-	}
-	mpl := mgr.getPL(streamID)
+	mpl := mgr.getPL(profile.Name)
 	if mpl == nil {
-		mpl = mgr.createPL(streamID)
+		mpl = mgr.createPL(profile)
```

Comment:
Fixed this in a separate branch since it's unrelated to failover. PR coming shortly. https://github.com/livepeer/go-livepeer/compare/ja/failover...ja/playlist